h2/src/main/org/h2/mvstore/FileStore.java

```diff
@@ -339,11 +339,12 @@ public void markUsed(long pos, int length) {
     }
 
     /**
-     * Allocate an area of the file and mark it as used.
+     * Allocate a number of blocks and mark them as used.
      *
      * @param length the number of bytes to allocate
-     * @param reservedLow start position of the reserved are
-     * @param reservedHigh end position of the reserved area
+     * @param reservedLow start block index of the reserved area (inclusive)
+     * @param reservedHigh end block index of the reserved area (exclusive),
+     *                     special value -1 means beginning of the infinite free area
      * @return the start position in bytes
      */
     long allocate(int length, long reservedLow, long reservedHigh) {
```

Comment:
Proposing use of the type **int** for **block index** instead of type long.

Reply:
I wasn't the one who picked it to be long. Choosing int will limit file size (with current block size) to 16TB. I think we can live with that, given that BitSet-based implementation imposes that int limit anyway.

Reply:
In the past, those parameters represented file position instead of block index, but now you have changed it to block index.

Reply:
No, I am talking about type of **Chunk.block** field. That's where all this "long" story starts.